core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/messaginglayer/OutputManager.scala

```diff
@@ -142,6 +152,25 @@ class OutputManager(
     }
   }
 
+  def saveTupleToStorageIfNeeded(
```

Comment:
please also add documentation/comments. it is not straightforward what is the expected behavior with different inputs.

Reply:
Added.